src/azul/indexer.py

```diff
@@ -176,8 +196,60 @@ def merge(cur_bundles, new_bundles):
             except KeyError:
                 bundle = new_bundle
             else:
-                bundle = new_bundle if new_bundle['version'] >= cur_bundle['version'] else cur_bundle
+                bundle = new_bundle if ((new_bundle['version'] >= cur_bundle['version']) or
+                                        new_bundle['admin_deleted']) else cur_bundle
             assert bundles.setdefault(bundle_uuid, bundle) is bundle
         for bundle in cur_bundles_by_id.values():
             assert bundles.setdefault(bundle['uuid'], bundle) is bundle
         return list(bundles.values())
+
+    def delete(self, dss_notification: Mapping[str, Any]) -> None:
+        bundle_uuid = dss_notification['match']['bundle_uuid']
+        bundle_version = dss_notification['match']['bundle_version']
+
+        es_client = self.properties.elastic_search_client
+
+        search_query = {
+            "query": {
+                "bool": {
+                    "must": [
+                        {
+                            "term": {
+                                "bundles.uuid.keyword": {
+                                    "value": bundle_uuid
+                                }
+                            }
+                        },
+                        {
+                            "term": {
+                                "bundles.version.keyword": {
+                                    "value": bundle_version
+                                }
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+
+        current_doc = es_client.search(doc_type="doc",
+                                       body=search_query)
+
+        hits = current_doc['hits']['hits']
+
+        updated_hits = []
+
+        for hit in hits:
+            current_doc_id = hit['_id']
+            hit_doc = es_client.get(index=hit['_index'],
```

Comment:
Why don't you fetch the source with query? This 1+N approach is really inefficient.